### ISSUE TITLE
jp2 support

### ISSUE THREAD

**Issue by 03302024** [GyanD/codexffmpeg#152]
I use Icaros (https://github.com/Xanashi/Icaros) for thumbnail generation in Windows 11. It uses ffmpeg for that. Icaros has trouble with some jp2 files. I reported the issue here: [ jp2 support #108 ](https://github.com/Xanashi/Icaros/issues/108) and got a response that it is ffmpeg's fault.

I am not that knowledgeable to explain the issue in technical terms, but in plain English, I get black thumbnails for these sample files:
1) https://archive.org/download/dr_pratts-high-test-plan-of-the-eastern-counties-and-the-midlands-12471000/12471000.jp2
2) [test file.zip](https://github.com/user-attachments/files/16325531/test.file.zip)

Can you please have a look?

**Comment by GyanD** [comment]
The first file cannot be decoded by either ffmpeg's native jpeg2000 decoder

```
SOC marker not present
error during processing marker segment ff55
```

or by the libopenjpeg decoder in an older version of ffmpeg

```
Expected a SOC marker
Error decoding codestream header.
Empty SOT marker detected: Psot=12.
```

So this seems like an issue with the file. 

The 2nd file can be decoded by libopenjpeg but not by the native decoder as it uses a palette  For the 2nd type of file, use an older ffmpeg build (5.1) with the libopenjpeg.

**Comment by 03302024** [comment]
Thank you for looking into the issues. 

There are more jp2 files on archive.org website with the same issue. XnViewMP and IrfanView can show the files OK. 

So, may be the errors are not severe and ffmpeg could just treat the errors as warnings and move on?

**Comment by GyanD** [comment]
That has to be be addressed by the native jpeg2000 decoder maintainer. Open a ticket at trac.ffmpeg.org for that.

**Comment by 03302024** [comment]
OK, thank you.
